cl/disclosures/tasks.py

```diff
@@ -216,142 +224,180 @@ def save_disclosure(extracted_data: dict, disclosure) -> None:
             )
-            for reimbursement in extracted_data["sections"]["Reimbursements"][
-                "rows"
-            ]
-        ]
-    )
 
-    NonInvestmentIncome.objects.bulk_create(
-        [
-            NonInvestmentIncome(
-                financial_disclosure=disclosure,
-                redacted=any(
-                    v["is_redacted"] for v in non_investment_income.values()
-                ),
-                date_raw=non_investment_income["Date"]["text"],
-                source_type=non_investment_income["Source and Type"]["text"],
-                income_amount=non_investment_income["Income"]["text"],
+            NonInvestmentIncome.objects.bulk_create(
+                [
+                    NonInvestmentIncome(
+                        financial_disclosure=disclosure,
+                        redacted=any(
+                            v["is_redacted"]
+                            for v in non_investment_income.values()
+                        ),
+                        date_raw=non_investment_income["Date"]["text"],
+                        source_type=non_investment_income["Source and Type"][
+                            "text"
+                        ],
+                        income_amount=non_investment_income["Income"]["text"],
+                    )
+                    for non_investment_income in extracted_data["sections"][
+                        "Non-Investment Income"
+                    ]["rows"]
+                ]
             )
-            for non_investment_income in extracted_data["sections"][
-                "Non-Investment Income"
-            ]["rows"]
-        ]
-    )
 
-    SpouseIncome.objects.bulk_create(
-        [
-            SpouseIncome(
-                financial_disclosure=disclosure,
-                redacted=any(v["is_redacted"] for v in spouse_income.values()),
-                date_raw=spouse_income["Date"]["text"],
-                source_type=spouse_income["Source and Type"]["text"],
+            SpouseIncome.objects.bulk_create(
+                [
+                    SpouseIncome(
+                        financial_disclosure=disclosure,
+                        redacted=any(
+                            v["is_redacted"] for v in spouse_income.values()
+                        ),
+                        date_raw=spouse_income["Date"]["text"],
+                        source_type=spouse_income["Source and Type"]["text"],
+                    )
+                    for spouse_income in extracted_data["sections"][
+                        "Non Investment Income Spouse"
+                    ]["rows"]
+                ]
             )
-            for spouse_income in extracted_data["sections"][
-                "Non Investment Income Spouse"
-            ]["rows"]
-        ]
-    )
+    except Exception as e:
+        logger.error(
+            msg=f"Error saving disclosure for {disclosure.person}",
+            extra=dict(error_msg=str(e)),
+        )
```

Comment:
Hm, you've got two code-smells here. The first is a really long try/except. The second is a too broad exception. Why is this all needed? Wouldn't the task just raise an exception otherwise that'd wind up in Sentry with the same info?

Reply:
I get that. I suppose these errors are already showing in sentry, so I can probably drop this one.